packages/cicero-core/src/templatelibrary.js

```diff
@@ -40,10 +40,19 @@ class TemplateLibrary {
      * Create the Template Library
      * @param {string} url - the url to connect to. Defaults to
      * https://templates.accordproject.org
+     * @param {object} auth - authenticate object
+     * @param {object}  auth.type - HTTP Auth type
```

Comment:
What's the benefit of splitting the HTTP header into `type` and `credential`, please? 

For example, the user still has the inconvenience of generating the base64 encoded string themselves.

Two ideas:
1. we allow users to provide the whole header themselves. This would allow us to support more complex auth schemes such as OAuth or AWS signatures where the credentials are generated by third-party code e.g. 
```
new TemplateLibrary(uri, { authorizationHeader: '...' });
```
2. We provide some nice abstraction for common uses (such as what `axios` does, https://github.com/axios/axios#request-config). This would configure the header to include a `Basic ` prefix and automatically base64 encode the credentials e.g.
```
new TemplateLibrary(uri, { username: '...', password: '...' });
```

We could do both? In which case the `authorizationHeader` would take precedence over the basic auth option.

Reply:
@mttrbrts I would like to go with the first option.Since it would allow `cli` to interact with `third party code.` Seond would restrict it.Plus both are going to give `authorizationHeader` would take precedence over the basic auth option.

Reply:
That sounds reasonable to me 👍🏻

Reply:
Perhaps a better name for this property is `httpHeader`?

Reply:
Sure I will make the required changes as told in the issue and pr comments.

Reply:
@mttrbrts What about Bearer Token,API Keys,Digest Auth and other type of Authentications.Shall I only test for Basic Auth with some credentials.

Reply:
Yes, that's fine. You're not testing the HTTP library per-say, only that the HTTP header is propagated through to the library.